internal/future/app/mmlogic/mmlogic_service.go

```diff
@@ -43,21 +55,51 @@ func newMmlogic(cfg config.View) (*mmlogicService, error) {
 	return ms, nil
 }
 
-// GetPoolTickets gets the list of Tickets that match every Filter in the
+// QueryTickets gets the list of Tickets that match every Filter in the
 // specified Pool.
-func (s *mmlogicService) QueryTickets(req *pb.QueryTicketsRequest, stream pb.MmLogic_QueryTicketsServer) error {
-	ctx := stream.Context()
+func (s *mmlogicService) QueryTickets(req *pb.QueryTicketsRequest, responseServer pb.MmLogic_QueryTicketsServer) error {
+	poolFilters := req.Pool.Filter
+
+	// Send requests to the storage service
+	idsToProperties, err := s.store.FilterTickets(poolFilters)
+	if err != nil {
+		logger.WithError(err).Error("Failed to retrieve result from storage service.")
+		return status.Error(codes.Internal, err.Error())
+	}
 
-	for {
+	ctx := responseServer.Context()
+	idx := 0
+	// The number of tickets in a paging response
+	pSize := s.cfg.GetInt("storage.page.size")
+	page := []*pb.Ticket{}
+
+	for id, property := range idsToProperties {
 		select {
 		case <-ctx.Done():
 			return nil
-
 		default:
-			err := stream.Send(&pb.QueryTicketsResponse{})
+			propertyByte, err := json.Marshal(property)
 			if err != nil {
-				return err
+				logger.WithError(err).Error("Failed to convert go struct to JSON")
+				return status.Errorf(codes.Internal, err.Error())
+			}
+			propertyJSON := string(propertyByte)
+			page = append(page, &pb.Ticket{Id: id, Properties: propertyJSON})
```

Comment:
nit: simply use string(propertyByte) here? no need for extra variable.

Reply:
I think it might be worth to create self-explained variable names to improve code readability, but you are right!